File: backend/spring-boot/src/main/java/org/bootstrapbugz/backend/auth/jwt/service/RefreshTokenService.java
package org.bootstrapbugz.backend.auth.jwt.service;

import java.util.Optional;
import java.util.Set;
import org.bootstrapbugz.backend.user.payload.dto.RoleDTO;

public interface RefreshTokenService {
  String create(Long userId, Set<RoleDTO> roleDTOs, String ipAddress);

  void check(String token);

  Optional<String> findByUserIdAndIpAddress(Long userId, String ipAddress);

  void delete(String token);

  void deleteByUserIdAndIpAddress(Long userId, String ipAddress);

  void deleteAllByUserId(Long userId);
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/jwt/service/RefreshTokenService.kt
package org.bootstrapbugz.api.auth.jwt.service

import java.util.Optional
import org.bootstrapbugz.api.user.payload.dto.RoleDTO

interface RefreshTokenService {
  fun create(userId: Long?, roleDTOs: Set<RoleDTO>, ipAddress: String): String

  fun check(token: String)

  fun findByUserIdAndIpAddress(userId: Long?, ipAddress: String): Optional<String>

  fun delete(token: String)

  fun deleteByUserIdAndIpAddress(userId: Long?, ipAddress: String)

  fun deleteAllByUserId(userId: Long?)
}
